### PR TITLE
Improve packet type

### PR DESCRIPTION
**Pull request by naseemkullah**
### The kind of change this PR does introduce

* [ ] a bug fix
* [ ] a new feature
* [x] an update to the documentation (formatted readme)
* [ ] a code change that improves performance
* [x] other

### Other information (e.g. related issues)

improve packet type

### CODE REVIEW

**Comment by naseemkullah** [comment]
cc @darrachequesne

File: lib/socket.ts [hunk 1]
```diff
@@ -6,6 +6,14 @@ import { Server } from "./server";
 
 const debug = debugModule("engine:socket");
 
+type PacketType = "error" | "message" | "open" | "ping" | "pong";
+
+export interface Packet {
+  type: PacketType;
+  options: { compress: boolean };
+  data?: any;
```
Comment (naseemkullah) [inline lib/socket.ts:14]: @darrachequesne  I noticed within the engine codebase `string` is sufficient, but IRL i also got array type, can this really be anything?

**Comment by naseemkullah** [comment]
CI failing with:


```
npm ERR! Error while executing:
npm ERR! /usr/bin/git ls-remote -h -t ssh://git@github.com/uNetworking/uWebSockets.js.git
npm ERR! 
npm ERR! Warning: Permanently added the RSA host key for IP address '140.82.112.4' to the list of known hosts.
npm ERR! git@github.com: Permission denied (publickey).
npm ERR! fatal: Could not read from remote repository.
npm ERR! 
npm ERR! Please make sure you have the correct access rights
npm ERR! and the repository exists.
npm ERR! 
npm ERR! exited with error code: 128

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/runner/.npm/_logs/2021-11-25T21_31_18_011Z-debug.log
Error: Process completed with exit code 1.
```

**Comment by darrachequesne** [comment]
@naseemkullah your PR has been merged in https://github.com/socketio/engine.io/commit/e122e4be7bc6b131552dba4dd4581a2ffb3797c6

Thanks a lot for your work on this :+1:

**Pull request closed without merging**